testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/DeployTestCase.java

```diff
@@ -73,84 +82,139 @@ public static void before() throws Exception {
         ctx.connectController();
-
-        String tempDir = System.getProperty("java.io.tmpdir");
+        AbstractCliTestBase.initCLI(TestSuiteEnvironment.getServerAddress());
 
         // deployment1
-        WebArchive war = ShrinkWrap.create(WebArchive.class, "cli-test-app1-deploy.war");
-        war.addAsWebResource(new StringAsset("Version0"), "page.html");
-        cliTestApp1War = new File(tempDir + File.separator + war.getName());
-        new ZipExporterImpl(war).exportTo(cliTestApp1War, true);
+        cliTestApp1War = createWarArchive("cli-test-app1-deploy.war", "Version0");
 
         // deployment2
-        war = ShrinkWrap.create(WebArchive.class, "cli-test-app2-deploy.war");
-        war.addAsWebResource(new StringAsset("Version1"), "page.html");
-        cliTestApp2War = new File(tempDir + File.separator + war.getName());
-        new ZipExporterImpl(war).exportTo(cliTestApp2War, true);
+        cliTestApp2War = createWarArchive("cli-test-app2-deploy.war", "Version1");
 
         // deployment3
-        war = ShrinkWrap.create(WebArchive.class, "cli-test-another-deploy.war");
-        war.addAsWebResource(new StringAsset("Version2"), "page.html");
-        cliTestAnotherWar = new File(tempDir + File.separator + war.getName());
-        new ZipExporterImpl(war).exportTo(cliTestAnotherWar, true);
-
-        ctx.handle("deploy --disabled " + cliTestApp1War.getAbsolutePath());
-        ctx.handle("deploy --disabled " + cliTestAnotherWar.getAbsolutePath());
-        ctx.handle("deploy --disabled " + cliTestApp2War.getAbsolutePath());
-
+        cliTestAnotherWar = createWarArchive("cli-test-another-deploy.war", "Version2");
     }
 
     @AfterClass
     public static void after() throws Exception {
-        ctx.handle("undeploy *");
         ctx.terminateSession();
+        AbstractCliTestBase.closeCLI();
+
         cliTestApp1War.delete();
         cliTestApp2War.delete();
         cliTestAnotherWar.delete();
     }
 
-    @Before
-    public void beforeTest() throws Exception {
-        if (readDeploymentStatus(cliTestApp1War.getName())) {
-            ctx.handle("deployment disable " + cliTestApp1War.getName());
-        }
-        if (readDeploymentStatus(cliTestAnotherWar.getName())) {
-            ctx.handle("deployment disable " + cliTestAnotherWar.getName());
+    @After
+    public void afterTest() {
+        ctx.handleSafe("deployment undeploy *");
+        if (tempCliTestAppWar != null) {
+            tempCliTestAppWar.delete();
         }
-        if (readDeploymentStatus(cliTestApp2War.getName())) {
-            ctx.handle("deployment disable " + cliTestApp2War.getName());
-        }
-        checkDeployment(cliTestApp1War.getName(), false);
-        checkDeployment(cliTestAnotherWar.getName(), false);
-        checkDeployment(cliTestApp2War.getName(), false);
     }
 
+    /**
+     * Test verify a live cycle of deployment operation in singleton mode.
```

Comment:
live cycle ==> life cycle. Could apply to other parts.

Reply:
Oh. Thank you. Done.